src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -219,7 +219,7 @@ private static bool TryGetByteArrayFromFileStream(Stream stream, out byte[] buff
             // than the buffer size. The default buffer size is 4KB, so this will incur a 4KB
             // allocation for any files less than 4KB. That's why, for example, the command
             // line compiler actually specifies a very small buffer size.
-            return stream.Read(buffer, 0, length) == length;
+            return stream.TryReadAll(buffer, 0, length) == length;
```

Comment:
I'm not sure if forcing more reads is in-line with the perf goals set out for this method. @pharring, would this produce additional allocations that we were trying to avoid?